autopush/db.py

```diff
@@ -649,6 +827,11 @@ def register_user(self, data):
                         r[key] = value
                 result = r
             return (True, result)
+        except ClientError as ex:
+            # ClientErrors are generated by a factory, and while they have a
+            # class, it's dynamically generated.
+            if 'ConditionalCheckFailedException' == ex.__class__.__name__:
```

Comment:
Since they recommend checking against `[Error']['Code']` like you did above for ResourceNotFoundException, let's just standardize on that. The test can raise ClientError({'Error': {'Code': 'ConditionalCheckFailedException'}})